forth_src/base.fs

```diff
@@ -1,15 +1,16 @@
 : 2+ 1+ 1+ ;
 : nip swap drop ;
 : jmp, 4c c, ;
-: ['] ' [ ' literal compile, ]
+: postpone bl word dup find ?dup 0= if
+count rvs type '?' emit cr abort then
+rot drop -1 = if [ ' literal compile,
+' compile, literal ] then compile,
```

Comment:
At this point, I don't really understand what the `' literal compile, ` compile, literal` twister does. But it seems to work fine.

Reply:
It's still the same two logical parts. The first three words were already there, and the last three replace `['] compile,`; the brackets perform things at compile time of `postpone`, it still looks up `compile,`, and `literal` transfers the result from the compile time stack to the run time stack. The `'` and `literal` do the same job as inside the compiled `[']`. I do agree it's a little boggling to see the long string of 3 words 7 times in rapid succession. It's because we don't yet have `postpone` and `[']`; the block would read `if postpone literal ['] compile, then` with them.

Reply:
I think I get it now.